### PR TITLE
XTCE generator scripts

### LINKED ISSUE #576: Improvements to XTCE generator script(s)
We have scripts to generate XTCE-formatted XML files for each instrument (example [here](https://github.com/IMAP-Science-Operations-Center/imap_processing/blob/dev/tools/xtce_generation/xtce_generator_codice.py)). These are used to generate the ["packet definition files"](https://github.com/IMAP-Science-Operations-Center/imap_processing/tree/dev/imap_processing/codice/packet_definitions) which are used to decommutate packet data from the spacecraft.

There are some improvements we could make to these scripts:
- Make a single, generalized script that can process any instrument, instead of having a script for each instrument. This could be done by passing in command line argument(s) for the instrument to process, the path to the user's telemetry spreadsheet, the path to which to write the packet definition file, etc.
- Improve the information that gets printed to the terminal for the user. Currently it is pretty bare bones.
- Add appropriate unit tests

### PR DESCRIPTION
**Pull request by anamanica**
# Change Summary

## Overview
Updated the xtce_generatpr_template.py to use CLI for any instrument.

## New Dependencies
None

## New Files
- test_xtce_generator_template
    - Wrote a few quick unit tests to ensure CLI is working.

## Deleted Files
None

## Updated Files
- xtce_generator_template
    - Finished implementation to take CLI and generate an xtce file.
- pyproject.toml
    - Added imap_xtce as a tool

## Testing
Three unit tests present in test_xtce_generator_template. Fairly general, but effective.

closes #576

### CODE REVIEW

File: tools/xtce_generation/xtce_generator_template.py
Comment (maxinelasp) [inline tools/xtce_generation/xtce_generator_template.py]: Could you rename this to "file_name"?

File: tools/xtce_generation/xtce_generator_template.py
Comment (maxinelasp) [inline tools/xtce_generation/xtce_generator_template.py]: Use `pathlib` here since windows uses `\`

File: tools/xtce_generation/xtce_generator_template.py
Comment (maxinelasp) [inline tools/xtce_generation/xtce_generator_template.py:73]: Could you also validate that the file exists?

**Review by maxinelasp — COMMENTED** [review]
Nice, looking great. The tests look awesome!

File: tools/xtce_generation/xtce_generator_template.py
Comment (sdhoyt) [inline tools/xtce_generation/xtce_generator_template.py]: Interested to hear if @maxinelasp agrees, but I'd prefer the command argument be the full path to the excel file rather than needing to put the excel file in the xtce_generation folder. I don't think we want to store the excel sheets in the repo so it's an extra step to copy one in there to run this command.

**Review by sdhoyt — COMMENTED** [review]
Great work on this!

File: tools/xtce_generation/xtce_generator_template.py
Comment (maxinelasp) [reply tools/xtce_generation/xtce_generator_template.py]: Yep, I agree, good call

Comment (anamanica) [reply tools/xtce_generation/xtce_generator_template.py]: Good stuff! I'll get that implemented, and then request another review.

File: imap_processing/tests/hi/test_hi_l1b.py
Comment (greglucas) [inline imap_processing/tests/hi/test_hi_l1b.py:1]: I think this was fixed in https://github.com/IMAP-Science-Operations-Center/imap_processing/pull/681 so shouldn't be added back in here.

File: tools/tests/test_xtce_generator_template.py
Comment (greglucas) [inline tools/tests/test_xtce_generator_template.py]: I'd suggest removing your specific folder structure from tests if possible. That way the bots won't learn who you are :)
```suggestion
        "tools/xtce_generation/TLM_SWP_20231006-121021.xlsx",
```

File: tools/xtce_generation/xtce_generator_template.py
Comment (greglucas) [inline tools/xtce_generation/xtce_generator_template.py:9]: I am not sure that this will work as an external script, were you able to verify that this works from other directories? We aren't installing the tools folder, so this currently has to be referenced directly from the folder structure I think.

File: tools/xtce_generation/xtce_generator_template.py
Comment (greglucas) [inline tools/xtce_generation/xtce_generator_template.py]: It should be hyphens for the CLI input to be consistent with our other CLIs. That then gets transformed to underscores within the Python code for you.
```suggestion
    parser.add_argument("--file-path", type=str, required=True, help=file_path_help)
```

File: tools/xtce_generation/xtce_generator_template.py
Comment (greglucas) [inline tools/xtce_generation/xtce_generator_template.py:93]: I was confused at first where the output was going, could you add a note in the main docstring about writing the file to the specific instrument subfolder?

**Review by greglucas — COMMENTED** [review]
Just a note that I have a proposal to rework some of the underlying XTCE code: https://github.com/IMAP-Science-Operations-Center/imap_processing/pull/654 if you wanted to review that as well. I think this PR will slot in nicely with it to enable a command line interface for that, but it might remove some of the arguments here (packets for instance).

**Review by maxinelasp — COMMENTED** [review]
Looks great to me, once you address Greg's comments which I agree with :)

**Review by sdhoyt — COMMENTED** [review]
Looks good to me! Thanks for making that update. Ready to approve once Greg's comments are addressed.

File: imap_processing/tests/hi/test_hi_l1b.py
Comment (anamanica) [reply imap_processing/tests/hi/test_hi_l1b.py:1]: Haha, yes! sorry about that.

File: tools/xtce_generation/xtce_generator_template.py
Comment (anamanica) [reply tools/xtce_generation/xtce_generator_template.py:9]: @greglucas That makes sense! In the external libraries, `imap-data-access` is loaded in, and this is a line of code I totally stole from `cli.py`. Would that make this okay to keep as is?

I tried to download the imap-xtce in a different directory to test it, but was unable to. Whether or not this is user error or actual error I am unsure, so I reached out to the students slack-chat.

Comment (greglucas) [reply tools/xtce_generation/xtce_generator_template.py:9]: Thanks for looking into this!

I think this is only an issue here because it is in the "tools" area, which is an extra portion and not something that gets added to the distribution when we type `pip install mypackage`. The way around this would be to inline `TelemetryGenerator`, or put this CLI work into the same file as that so there are no relative imports required.

I wouldn't bother with this. We can know about the limitation and this is an improvement 👍 . That can be added/updated as an extra feature later.

**Review by greglucas — APPROVED** [review]

**Pull request merged**
